components/kyma-environment-broker/internal/broker/instance_last_operation_test.go

```diff
@@ -56,6 +57,127 @@ func TestLastOperation_LastOperation(t *testing.T) {
+		updateOp := fixture.FixUpdatingOperation(operationID, instID)
+		updateOp.State = orchestration.Retrying
+		err := memoryStorage.Operations().InsertUpdatingOperation(updateOp)
+		assert.NoError(t, err)
+
+		lastOperationEndpoint := broker.NewLastOperation(memoryStorage.Operations(), logrus.StandardLogger())
+
+		// when
+		response, err := lastOperationEndpoint.LastOperation(context.TODO(), instID, domain.PollDetails{OperationData: ""})
+		assert.NoError(t, err)
+
+		// then
+		assert.Equal(t, domain.LastOperation{
+			State:       domain.InProgress,
+			Description: updateOp.Description,
+		}, response)
+
+		// when
+		response, err = lastOperationEndpoint.LastOperation(context.TODO(), instID,
+			domain.PollDetails{OperationData: operationID})
+		assert.NoError(t, err)
+
+		// then
+		assert.Equal(t, domain.LastOperation{
+			State:       domain.InProgress,
+			Description: updateOp.Description,
+		}, response)
+	})
+	t.Run("Should convert operation's canceling state to succeeded", func(t *testing.T) {
+		// given
+		memoryStorage := storage.NewMemoryStorage()
+		updateOp := fixture.FixUpdatingOperation(operationID, instID)
+		updateOp.State = orchestration.Canceling
+		err := memoryStorage.Operations().InsertUpdatingOperation(updateOp)
+		assert.NoError(t, err)
+
+		lastOperationEndpoint := broker.NewLastOperation(memoryStorage.Operations(), logrus.StandardLogger())
+
+		// when
+		response, err := lastOperationEndpoint.LastOperation(context.TODO(), instID, domain.PollDetails{OperationData: ""})
+		assert.NoError(t, err)
+
+		// then
+		assert.Equal(t, domain.LastOperation{
+			State:       domain.Succeeded,
+			Description: updateOp.Description,
+		}, response)
+
+		// when
+		response, err = lastOperationEndpoint.LastOperation(context.TODO(), instID,
+			domain.PollDetails{OperationData: operationID})
+		assert.NoError(t, err)
+
+		// then
+		assert.Equal(t, domain.LastOperation{
+			State:       domain.Succeeded,
+			Description: updateOp.Description,
+		}, response)
+	})
+	t.Run("Should convert operation's canceled state to succeeded", func(t *testing.T) {
+		// given
+		memoryStorage := storage.NewMemoryStorage()
+		updateOp := fixture.FixUpdatingOperation(operationID, instID)
+		updateOp.State = orchestration.Canceled
+		err := memoryStorage.Operations().InsertUpdatingOperation(updateOp)
+		assert.NoError(t, err)
+
+		lastOperationEndpoint := broker.NewLastOperation(memoryStorage.Operations(), logrus.StandardLogger())
+
+		// when
+		response, err := lastOperationEndpoint.LastOperation(context.TODO(), instID, domain.PollDetails{OperationData: ""})
+		assert.NoError(t, err)
+
+		// then
+		assert.Equal(t, domain.LastOperation{
+			State:       domain.Succeeded,
+			Description: updateOp.Description,
+		}, response)
+
+		// when
+		response, err = lastOperationEndpoint.LastOperation(context.TODO(), instID,
+			domain.PollDetails{OperationData: operationID})
+		assert.NoError(t, err)
+
+		// then
+		assert.Equal(t, domain.LastOperation{
```

Comment:
Does the second invocation change something?

Reply:
With operation id provided as query parameter last operation endpoint runs GetOperationById() instead of GetLastOperation(). There is a difference in queries between these two.